### ISSUE TITLE
Out of bounds mutables in static myclass$.<clinit>()V

### ISSUE THREAD

**Issue by fredxc** [find-sec-bugs/find-sec-bugs#199]
Thanks for the contribution, I am trying it with a couple of scala codes using FindSecBug Intellij plugin 1.0.0. An Assertion error is reported in back trace.

```
java.lang.AssertionError: Out of bounds mutables in static MyClass$.()V
at com.h3xstream.findsecbugs.taintanalysis.TaintFrameModelingVisitor.transferTaintToMutables(TaintFrameModelingVisitor.java:434)
at com.h3xstream.findsecbugs.taintanalysis.TaintFrameModelingVisitor.visitInvoke(TaintFrameModelingVisitor.java:280)
at com.h3xstream.findsecbugs.taintanalysis.TaintFrameModelingVisitor.visitINVOKESPECIAL(TaintFrameModelingVisitor.java:203)
at org.apache.bcel.generic.INVOKESPECIAL.accept(INVOKESPECIAL.java:80)
at edu.umd.cs.findbugs.ba.AbstractFrameModelingVisitor.analyzeInstruction(AbstractFrameModelingVisitor.java:84)
at com.h3xstream.findsecbugs.taintanalysis.TaintAnalysis.transferInstruction(TaintAnalysis.java:66)
at com.h3xstream.findsecbugs.taintanalysis.TaintAnalysis.transferInstruction(TaintAnalysis.java:38)
at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.transfer(AbstractDataflowAnalysis.java:135)
at edu.umd.cs.findbugs.ba.Dataflow.execute(Dataflow.java:376)
at com.h3xstream.findsecbugs.taintanalysis.TaintDataflowEngine.analyze(TaintDataflowEngine.java:104)
at com.h3xstream.findsecbugs.taintanalysis.TaintDataflowEngine.analyze(TaintDataflowEngine.java:43)
at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.analyzeMethod(AnalysisCache.java:369)
at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.getMethodAnalysis(AnalysisCache.java:322)
at com.h3xstream.findsecbugs.injection.AbstractTaintDetector.getTaintDataFlow(AbstractTaintDetector.java:107)
at com.h3xstream.findsecbugs.injection.AbstractTaintDetector.analyzeMethod(AbstractTaintDetector.java:75)
at com.h3xstream.findsecbugs.injection.AbstractTaintDetector.visitClassContext(AbstractTaintDetector.java:60)
at edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
at edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1089)
at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:283)
at org.twodividedbyzero.idea.findbugs.core.FindBugsStarter.executeImpl(FindBugsStarter.java:354)
at org.twodividedbyzero.idea.findbugs.core.FindBugsStarter.asyncStartImpl(FindBugsStarter.java:245)
at org.twodividedbyzero.idea.findbugs.core.FindBugsStarter.asyncStart(FindBugsStarter.java:218)
at org.twodividedbyzero.idea.findbugs.core.FindBugsStarter.access$300(FindBugsStarter.java:74)
at org.twodividedbyzero.idea.findbugs.core.FindBugsStarter$3.run(FindBugsStarter.java:202)
at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:446)
at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:126)
at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
```

**Comment by h3xstream** [comment]
This seems to be an identical error to #112

**Comment by h3xstream** [comment]
@fredxc Can you confirm that this error is still occuring?

I will need a sample to reproduce and the version of the plugin you used.
note: FindBugs-IDEA does not always have the latest version of FindSecurityBugs.

**Comment by h3xstream** [comment]
@fredxc I will close the issue assuming it is related to the race condition that was fixed (#112).

Don't hesitate to comment on this issue if your tests confirm that it is still present.
